sql/sql_prepare.cc

```diff
@@ -3156,9 +3157,16 @@ bool Prepared_statement::execute_loop(String *expanded_query,
       }
     }
 
-    if (!error) /* Success */
+    if (!error) { /* Success */
+#ifdef WITH_WSREP
+      // We are going to retry the statement, so clean up first.
+      wsrep_after_statement(thd);
+#endif
```

Comment:
Can we do it this way please?
```diff
     }
 
     if (!error) /* Success */
+#ifdef WITH_WSREP
+    // We are going to retry the statement, so clean up first.
+    {
+      wsrep_after_statement(thd);
+#endif /* WITH_WSREP */
       goto reexecute;
+#ifdef WITH_WSREP
+    }
+#endif /* WITH_WSREP */
   }
   reset_stmt_parameters(this);
```

Reply:
Right! That's the only way.
Added also missing WITH_WSREP around the include file.